Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -241,9 +264,16 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
           .materialize()
       }
 
-    self.validateCheckoutSuccess = selectedNewCreditCard
+    let validateCheckoutNewCardSuccess: Signal<PaymentSourceValidation, Never> = selectedCard
       .takeWhen(validateCheckoutNewCard.values())
-      .map { _, paymentIntentClientSecret, _ in paymentIntentClientSecret }
+      .map { paymentSource, _, _ in PaymentSourceValidation(
+        paymentIntentClientSecret: paymentSource.paymentIntentClientSecret!,
+        selectedCardStripeCardId: nil,
+        requiresConfirmation: false // Newly added cards are confirmed in PledgePaymentMethodsViewController
+      ) }
+
+    self.validateCheckoutSuccess = Signal
+      .merge(validateCheckoutNewCardSuccess, validateCheckoutExistingCardSuccess)
 
     self.showErrorBannerWithMessage = Signal
       .combineLatest(validateCheckoutExistingCard.errors(), validateCheckoutNewCard.errors())
```

Comment:
nit: Do you want to swap this to a `merge` while you're here?

Reply:
Will do